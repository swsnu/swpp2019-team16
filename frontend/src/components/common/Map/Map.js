import React from 'react';
import PropTypes from 'prop-types';
import GoogleMap from 'google-map-react';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles({
  root: {
    width: props => props.width,
    height: props => props.height,
  },
});

Map.propTypes = {
  width: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  center: PropTypes.shape({
    lat: PropTypes.number.isRequired,
    lng: PropTypes.number.isRequired,
  }).isRequired,
  zoom: PropTypes.number,
};

function Map({ width, height, children, center, zoom = 17 }) {
  const styles = useStyles({ width, height });
  return (
    <div className={styles.root}>
      <GoogleMap center={center} defaultZoom={zoom}>
        {children}
      </GoogleMap>
    </div>
  );
}

export default Map;
