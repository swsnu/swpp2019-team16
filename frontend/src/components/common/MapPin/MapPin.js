import React from 'react';
import { makeStyles } from '@material-ui/styles';
import PersonPin from '@material-ui/icons/PersonPin';
import IconButton from '@material-ui/core/IconButton';

const useStyles = makeStyles({
  root: {
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    transform: 'translate(-50%, -50%)',
  },
  icon: {
    width: '50px',
    height: '50px',
  }
});

function MapPin() {
  const styles = useStyles();
  return (
    <IconButton
      classes={{
        root: styles.root,
      }}
    >
      <PersonPin
        classes={{
          root: styles.icon,
        }}
        color={'secondary'}
      />
    </IconButton>
  );
};

export default MapPin;