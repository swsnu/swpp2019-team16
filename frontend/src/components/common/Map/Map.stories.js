import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { storiesOf } from '@storybook/react';

import Map from './Map';
import { ThemeProvider } from '@material-ui/styles';
import theme from '../../../lib/styles/theme';
import { CssBaseline } from '@material-ui/core';
import MapPin from '../MapPin/MapPin';

storiesOf('common/Map', module)
  .addDecorator(story => (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        {story()}
      </ThemeProvider>
    </BrowserRouter>
  ))
  .add('default', () => (
    <Map
      width={"100%"}
      height={"400px"}
      center={{
        lat: 37.480126,
        lng: 126.952436,
      }}
    >
      <MapPin
        lat={37.480126}
        lng={126.952436}
      />
    </Map>
  ));