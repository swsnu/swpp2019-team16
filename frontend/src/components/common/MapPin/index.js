import MapPin from './MapPin';
export default MapPin;