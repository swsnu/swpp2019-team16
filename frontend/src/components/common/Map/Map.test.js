import React from 'react';
import { render } from '@testing-library/react';
import Map from './Map';
import MapPin from '../MapPin';

describe('<Map />', () => {
  it('should match with snapshot - default', async () => {
    const { container } = render(
      <Map
        width={'100%'}
        height={'400px'}
        center={{
          lat: 37.480126,
          lng: 126.952436,
        }}
      >
        <MapPin lat={37.480126} lng={126.952436} />
      </Map>,
    );
    expect(container).toMatchSnapshot();
  });
});
