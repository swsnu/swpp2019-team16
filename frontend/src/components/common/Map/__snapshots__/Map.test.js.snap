// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> should match with snapshot - default 1`] = `
<div>
  <div
    class="makeStyles-root-1 makeStyles-root-2"
  >
    <div
      style="width: 100%; height: 100%; margin: 0px; padding: 0px; position: relative;"
    >
      <div
        style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
      />
      <div
        style="width: 50%; height: 50%; left: 50%; top: 50%; margin: 0px; padding: 0px; position: absolute;"
      >
        <div
          style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
        >
          <div
            style="width: 0px; height: 0px; left: 0px; top: 0px; background-color: transparent; position: absolute;"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root makeStyles-root-3"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon-4 MuiSvgIcon-colorSecondary"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2c-4.97 0-9 4.03-9 9 0 4.17 2.84 7.67 6.69 8.69L12 22l2.31-2.31C18.16 18.67 21 15.17 21 11c0-4.97-4.03-9-9-9zm0 2c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.3c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
